### PR TITLE
librdkafka: Bump to 1.7.0 (bump dependencies to their latest versions too)

### PR DESCRIPTION
**Pull request by alejandro-colomar**
**librdkafka/1.7.0**

Bump version

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by alejandro-colomar** [comment]
D'oh!  I edited the PR message, to see if the bot added the 'Bump version' tag, and it triggered a new build and reset the approvals.

**Comment by ericLemanissier** [comment]
It won't set the "Bump version" tag, because you modified conanfile.py, so all the versions need to be rebuilt. Also, I don't think this PR had any approval yet, so you didn't loose anything (except CI time).

File: recipes/librdkafka/all/conandata.yml [hunk 1]
```diff
@@ -17,36 +17,44 @@ sources:
...
-    - patch_file: patches/0001-Remove-cmake-module-path-and-change-library-names.patch
+    - patch_file: patches/0001-Remove-cmake-module-path-and-change-library-names-1-3-0.patch
       base_path: source_subfolder
     - patch_file: patches/0002-Change-library-targets-and-result-variables-1-4-0.patch
       base_path: source_subfolder
     - patch_file: patches/0003-Fix-broken-version-1-4-0-on-windows.patch
       base_path: source_subfolder
   "1.5.2":
-    - patch_file: patches/0004-Change-library-names.patch
+    - patch_file: patches/0001-Change-library-names-1-5-2.patch
```
Comment (prince-chrismc) [inline recipes/librdkafka/all/conandata.yml:37]: I'd prefer to see this change reverted

File: recipes/librdkafka/all/conandata.yml [hunk 2]
```diff
@@ -17,36 +17,44 @@ sources:
...
       base_path: source_subfolder
     - patch_file: patches/0002-Change-library-targets-and-result-variables-1-5-2.patch
       base_path: source_subfolder
   "1.5.3":
-    - patch_file: patches/0004-Change-library-names.patch
+    - patch_file: patches/0001-Change-library-names-1-5-2.patch
```
Comment (prince-chrismc) [inline recipes/librdkafka/all/conandata.yml:42]: Using a patch 1.5.2 on the "wrong version" it's why they have been just numbered

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

File: recipes/librdkafka/all/conandata.yml (hunk 2 above)
Comment (alejandro-colomar) [reply recipes/librdkafka/all/conandata.yml:42]: Hmm, I see.  A few thoughts:

- I agree that a specific version number in a patch name is a bit weird when it's applied to newer version, even though it makes sense if it still applies.
- Specifying a version range in the patch name is... unriendly
- numbering as 0004 the first patch is very weird.
- and of course, having identical patch files for every version is wrong.

What about having symlinks that apply to every version? I.e., for 1.5.3 there would be a symlink -1-5-3 that links to -1-5-2.

**Comment by alejandro-colomar** [comment]
Please review my comment above.  If you still prefer the old numbering, I'll revert that change.

File: recipes/librdkafka/all/conandata.yml (hunk 2 above)
Comment (prince-chrismc) [reply recipes/librdkafka/all/conandata.yml:42]: > numbering as 0004 the first patch feels wrong (IMO, more than applying a patch for an old version to a newer version)

🤔 Looking at the history there were patches removed as Conan gained features. We can reset the index if that's a compromise?

> What about having symlinks that apply to every version?

I am a little worried about symlinks being cross-platform, I am not sure how well handled that use case is... but excellent question!

Comment (alejandro-colomar) [reply recipes/librdkafka/all/conandata.yml:42]: I can do the symlinks and we'll see if the windows builds break...

**Review by prince-chrismc — APPROVED** [review]

File: recipes/librdkafka/all/conandata.yml (hunk 2 above)
Comment (prince-chrismc) [reply recipes/librdkafka/all/conandata.yml:42]: I approved, it's not an objective improvement... we'll see if anyone else is as picky as me today =)

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**